### ISSUE TITLE
Misuses of open flags

### ISSUE THREAD

**Issue by alb-i986** [AgNO3/jcifs-ng#223]
I also noticed that in `SmbFile#openInputStream` the constant `SmbConstants.O_RDONLY` is passed to the parameter `access`, instead of `sharing`. I don't think it was intended, was it?

See https://github.com/AgNO3/jcifs-ng/blob/59c3c0a89f541966f38dc50dd43d5e2a315e4fa3/src/main/java/jcifs/smb/SmbFile.java#L2123

From a quick search I spotted the same pattern in a couple of other places but possibly there may be more:
- https://github.com/AgNO3/jcifs-ng/blob/59c3c0a89f541966f38dc50dd43d5e2a315e4fa3/src/main/java/jcifs/smb/SmbCopyUtil.java#L120
- https://github.com/AgNO3/jcifs-ng/blob/59c3c0a89f541966f38dc50dd43d5e2a315e4fa3/src/main/java/jcifs/smb/SmbCopyUtil.java#L209

Please look into it.
Cheers

**Comment by mbechler** [comment]
This is meant to be the access mode, so no, I think this is correct. However, some cleanup of the constant uses may be a good idea to make this clearer.

**Comment by alb-i986** [comment]
Sorry, I got mixed up.
I meant that O_RDONLY is an _open flag_, so I think it should be passed as the first argument of openUnshared, which is "flags", instead of "access".

If you search for all the usages of O_RDONLY, you can see that some are using it the way I'm saying:

https://github.com/AgNO3/jcifs-ng/blob/593b67495f9c2ccd933d332b280cf642cbff2a59/src/main/java/jcifs/smb/SmbFile.java#L1042

https://github.com/AgNO3/jcifs-ng/blob/593b67495f9c2ccd933d332b280cf642cbff2a59/src/main/java/jcifs/smb/SmbRandomAccessFile.java#L116

But mostly it is passed as access.
Doesn't look good to me.

**Comment by mbechler** [comment]
Yes, right, this is off. Shouldn't hurt though (as long as the correct access is given) as _flags_ is only used to determine the create disposition. Better to get this consistent and get rid of the O_<access> constants.
